chain/client/src/info.rs

```diff
@@ -119,6 +120,57 @@ impl InfoHelper {
         metrics::FINAL_BLOCK_HEIGHT.set(last_final_block_height as i64);
         metrics::FINAL_DOOMSLUG_BLOCK_HEIGHT.set(last_final_ds_block_height as i64);
         metrics::EPOCH_HEIGHT.set(epoch_height as i64);
+        metrics::BLOCK_HEIGHT_WITHIN_EPOCH.set(block_height_within_epoch as i64);
+    }
+
+    fn record_tracked_shards(head: &Tip, client: &crate::client::Client) {
+        if let Ok(num_shards) = client.runtime_adapter.num_shards(&head.epoch_id) {
+            (0..num_shards).for_each(|shard_id| {
+                let tracked = client.runtime_adapter.cares_about_shard(
+                    None,
+                    &head.last_block_hash,
+                    shard_id,
+                    false,
+                );
+                metrics::TRACKED_SHARDS
+                    .with_label_values(&[shard_id.to_string().as_str()])
+                    .set(if tracked { 1 } else { 0 });
+            })
+        }
+    }
+
+    fn record_block_producers(head: &Tip, client: &crate::client::Client) {
+        let me = client.validator_signer.as_ref().map(|x| x.validator_id().clone());
+        let is_bp = me
+            .map(|account_id| {
+                client
+                    .runtime_adapter
+                    .get_epoch_block_producers_ordered(&head.epoch_id, &head.last_block_hash)
+                    .unwrap()
+                    .iter()
+                    .any(|bp| bp.0.account_id() == &account_id)
+            })
+            .unwrap_or(false);
+        metrics::IS_BLOCK_PRODUCER.set(if is_bp { 1 } else { 0 });
+    }
+
+    fn record_chunk_producers(head: &Tip, client: &crate::client::Client) {
+        let me = client.validator_signer.as_ref().map(|x| x.validator_id().clone());
+        me.map(|account_id| {
+            let epoch_info = client.runtime_adapter.get_epoch_info(&head.epoch_id);
+            epoch_info.map(|epoch_info| {
+                for (shard_id, validators) in
+                    epoch_info.chunk_producers_settlement().into_iter().enumerate()
+                {
+                    let is_chunk_producer_for_shard = validators.iter().any(|&validator_id| {
+                        epoch_info.validator_account_id(validator_id).clone() == account_id
```

Comment:
Do you need that clone just for the comparison?

Reply:
No, Thanks for catching this!